arch/common/include/sw_isr_common.h

```diff
@@ -0,0 +1,57 @@
+/*
```

Comment:
I've moved the helper functions into an internal header, is this what you meant? @nashif

Reply:
yeah, I may have overreacted to the sight of z_ APIs, it seems this header is full of them already. Need to spend some time on this and see what the status of the header and if it needs to be in the public domain in the first place. thanks.